### PR TITLE
[TD] Add Command3DLengthDimension a new command creating 3D length dimensions

### PR DESCRIPTION
**Pull request by edi271**
New command creating 3D length dimensions in TechDraw. This command does the same as the macro Create3DDimension031.FCMacro which was presented in https://forum.freecadweb.org/viewtopic.php?f=35&t=72114
Wiki will be updated in the next days.

Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [ ]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8300">pipeline status</a> for feature branch [PR_8300](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8300). Pipeline [761702993](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/761702993) was triggered at [393fce2](https://github.com/FreeCAD/FreeCAD/pull/8300/commits/393fce200abd9dc2f67669df29ed08fba9dc11c0). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by WandererFan** [comment]
Is this a "3d length dimension" or is it an isometric layout for a 2d dimension?  There doesn't seem to be any connection with 3d.

Really happy to see a python add-on!

**Comment by edi271** [comment]
> Is this a "3d length dimension" or is it an isometric layout for a 2d dimension? There doesn't seem to be any connection with 3d.
> 
> Really happy to see a python add-on!

It is an isometric layout for a 2d dimension. No connection to the really 3d values. The problem is not to calculate the 3d value. The problem is, that the calculated 3d value will be an arbitrary value, inconvenient to manage by the user.
I will publish in the next days in the forum a macro calculating the 3d values (if the measure is parallel to the main directions) by using the TechDraw view.

**Comment by WandererFan** [comment]
So, should we change the name before merging, or way for somebody to tell us we picked the wrong name? :)

File: src/Mod/TechDraw/TechDrawTools/Command3DLengthDimension.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                            vertexes.append(edges[0].Vertexes[1])
+                        return(edges,vertexes)
+                    else:
+                        return False
+                else:
+                    return False
+            else:
+                return False
+        else:
+            return False
```
Comment (benj5378) [inline src/Mod/TechDraw/TechDrawTools/Command3DLengthDimension.py:118]:
```suggestion
        if not Gui.Selection.getSelection():
            return False
        view = Gui.Selection.getSelection()[0]

        if not Gui.Selection.getSelectionEx():
            return False
        objectList = Gui.Selection.getSelectionEx()[0].SubElementNames

        if not len(objectList)>=2:
            return False

        edges = []
        vertexes = []
        for objectString in objectList:
            if objectString[0:4] == 'Edge':
                edges.append(view.getEdgeBySelection(objectString))
            elif objectString[0:6] == 'Vertex':
                vertexes.append(view.getVertexBySelection(objectString))
        if not len(edges) >= 2:

        if len(vertexes)<2:
            vertexes = []
            vertexes.append(edges[0].Vertexes[0])
            vertexes.append(edges[0].Vertexes[1])
        return(edges,vertexes)
```

bruh, do you need me to come back and preach

**Comment by edi271** [comment]
> So, should we change the name before merging, or way for somebody to tell us we picked the wrong name? :)

proposal: Command3DPlaneLengthDimension

**Comment by edi271** [comment]
> > So, should we change the name before merging, or way for somebody to tell us we picked the wrong name? :)
> 
> proposal: Command3DPlaneLengthDimension

or CommandPlane3DLengthDimension ?

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8300">pipeline status</a> for feature branch [PR_8300](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8300). Pipeline [765618934](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/765618934) was triggered at [547b8ba](https://github.com/FreeCAD/FreeCAD/pull/8300/commits/547b8ba139ecf3d534d986d87e101118141c1763). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by WandererFan** [comment]
see https://github.com/WandererFan/FreeCAD/tree/command3dlengthdimension

re-inserted missing def for "selectionTest"
changed name to CommandAxoLengthDimension
changed icon to remove "3D" text
edited tooltip
rebased on master

if you are ok with my changes, we should be good to merge.

**Comment by edi271** [comment]
> see https://github.com/WandererFan/FreeCAD/tree/command3dlengthdimension
> 
> re-inserted missing def for "selectionTest" changed name to CommandAxoLengthDimension changed icon to remove "3D" text edited tooltip rebased on master
> 
> if you are ok with my changes, we should be good to merge.

Thank you, for me everything is OK, please merge.

**Comment by WandererFan** [comment]
Replaced by #8359.  Closing.

**Pull request closed without merging**